ad_rando/seed_generator.py

```diff
@@ -8,18 +10,37 @@ def __init__(self, error_message):
 
 
 class SeedGenerator:
+    def __init__(self, validator):
+        self._validator = validator
+        self._seed_base = time.time() * 1000
+        self._seed_floor = math.floor(self._seed_base)
+        # Use the nanosecond portion of the time as a pseudo-random offset, plus a constant in case that happens to be 0
+        self._offset = math.floor((self._seed_base - self._seed_floor) * 1000) + 50
+        self._index = 0
+
+    def next(self):
+        seed = self._generate()
+        while not self._validator.validate(seed):
```

Comment:
Not important now, but may want to consider a timeout number of tries, so that if people add new presets and complex validators in the future they don't get into an infinite (or too long) loop.